pkgs/dart_mcp/lib/src/server/resources_support.dart

```diff
@@ -0,0 +1,168 @@
+// Copyright (c) 2025, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+part of 'server.dart';
+
+/// A mixin for MCP servers which support the `resources` capability.
+///
+/// Servers should add resources using the [addResource] method, typically
+/// inside the [initialize] method, but they may also be added after
+/// initialization if needed.
+///
+/// Resources can later be removed using [removeResource], or the client can be
+/// notified of updates using [updateResource].
+///
+/// Clients will be notified automatically of resource changes.
```

Comment:
do clients have to subscript to the resources/updated stream for this to be true?

Reply:
yes - reworded this